lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -121,12 +121,13 @@ IRGenDebugInfo::IRGenDebugInfo(const IRGenOptions &Opts,
   }
 
   unsigned Lang = llvm::dwarf::DW_LANG_Swift;
-  std::string Producer = version::getSwiftFullVersion();
+  std::string Producer = version::getSwiftFullVersion(
+    Opts.EffectiveLanguageVersion);
   bool IsOptimized = Opts.Optimize;
   StringRef Flags = Opts.DWARFDebugFlags;
-  unsigned Major, Minor;
-  std::tie(Major, Minor) = version::getSwiftNumericVersion();
-  unsigned MajorRuntimeVersion = Major;
+
+  assert(!Opts.EffectiveLanguageVersion.empty());
```

Comment:
Yes, my understanding is that we always want to emit the current swift runtime version here, even if the source the object was compiled from had a lower effective language version.